MyApplication/app/src/main/java/com/example/myapplication/adapter/ItemAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.myapplication.adapter
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myapplication.R
+import com.example.myapplication.model.Post
+
+class ItemAdapter(
+    private val context: Context,
+    private val dataset: List<Post>
+    ): RecyclerView.Adapter<ItemAdapter.ItemViewHolder>(){
+
+    class ItemViewHolder(private val view: View): RecyclerView.ViewHolder(view) {
+        val textView_2: TextView = view.findViewById(R.id.item_author)
+        val textView: TextView = view.findViewById(R.id.item_title)
+        val imageView: ImageView = view.findViewById(R.id.item_image)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
+        // create a new view
+        val adapterLayout = LayoutInflater.from(parent.context)
+            .inflate(R.layout.list_item, parent, false)
+
+        return ItemViewHolder(adapterLayout)
```

Comment:
Change init of recycler view item to view binding https://developer.android.com/topic/libraries/view-binding